assets/src/edit-story/components/keyboardShortcutsMenu/index.js

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useRef, useState, useCallback, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { useGlobalKeyDownEffect, useKeyDownEffect } from '../keyboard';
+import WithTooltip from '../tooltip';
+import Popup, { Placement } from '../popup';
+import { Keyboard as KeyboardShortcutsButton } from '../button';
+import ShortcutMenu from './shortcutMenu';
+
+function KeyboardShortcutsMenu({
+  placement = Placement.TOP_END,
+  spacing = { x: 0, y: 12 },
+  onMenuToggled,
+}) {
+  const anchorRef = useRef();
+  const isOpenBuffer = useRef(false);
+  const isOpenSetTimeout = useRef(0);
+
+  const [isOpen, setIsOpen] = useState(false);
+
+  const tryToggleMenu = useCallback(
+    (showMenu) => {
+      isOpenBuffer.current = showMenu;
+
+      // Stop previous timeout
+      window.clearTimeout(isOpenSetTimeout.current);
+
+      // Some toggle menu events can come in rapidly, to
+      // ensure that we don't just toggle the menu closed
+      // and then immediately open it, this little setTimeout
+      // will make sure only the last toggle will actually
+      // happen.
+      isOpenSetTimeout.current = window.setTimeout(() => {
+        setIsOpen(isOpenBuffer.current);
+        if (onMenuToggled) {
+          onMenuToggled(isOpenBuffer.current);
+        }
+
+        isOpenBuffer.current = false;
+      }, 10);
+    },
+    [onMenuToggled]
+  );
```

Comment:
This `setTimeout` is so that you can press the keyboard shortcut button to open _and_ close the menu.  Without it, when you click on the button the `focusOut` triggers the menu to close, and immediately after the `onClick` of the button gets processed and... re-opens the menu.  

The better solutions is probably to allow the `focusOut` hook to accept "ignore" containers or something, and then it won't trigger it's callback if something outside the container is clicked but within the "ignore" list...  But, I needed  quicker solution so I went with this.